src/platform/elements/date-picker/DatePickerInput.ts

```diff
@@ -109,32 +107,65 @@ export class NovoDatePickerInputElement implements OnInit, ControlValueAccessor
     }
   }
 
+  _handleBlur(event: FocusEvent): void {
+    let value = (event.target as HTMLInputElement).value;
+    try {
+      let dateTimeValue = Date.parse(value);
+      if (!isNaN(dateTimeValue)) {
+        let dt = new Date(dateTimeValue);
+        this.dispatchOnChange(dt, true);
+      } else {
+        this.dispatchOnChange(null, true);
+      }
+    } catch (err) {}
+    this.openPanel();
+  }
+
   writeValue(value: any): void {
     Promise.resolve(null).then(() => this._setTriggerValue(value));
   }
+
+  writeCalendarValue(value: any): void {
+    Promise.resolve(null).then(() => this._setCalendarValue(value));
+  }
+
   registerOnChange(fn: (value: any) => {}): void {
     this._onChange = fn;
   }
   registerOnTouched(fn: () => {}) {
     this._onTouched = fn;
   }
-  public dispatchOnChange(newValue?: any, skip: boolean = false) {
+  public dispatchOnChange(newValue?: any, blur: boolean = false, skip: boolean = false) {
     if (newValue !== this.value) {
-      this._onChange(newValue);
-      !skip && this.writeValue(newValue);
+      if (blur) {
+        this._onChange(newValue);
+        !skip && this.writeValue(newValue);
+      } else {
+        this._onChange(newValue);
+        !skip && this.writeCalendarValue(newValue);
+      }
     }
   }
 
   private _setTriggerValue(value: any): void {
+    this._setCalendarValue(value);
+    this._setFormValue(value);
+    this._changeDetectorRef.markForCheck();
+  }
+
+  private _setCalendarValue(value: any): void {
     if (value instanceof Date && this.value instanceof Date) {
       value = new Date(value.setHours(this.value.getHours(), this.value.getMinutes()));
     }
     this.value = value;
+    this._changeDetectorRef.markForCheck();
```

Comment:
Since you call this in setTriggerValue, you should probably move this to be called in the dispatchOnChange.

Reply:
resolved